mass_mailing_switzerland/models/mass_mailing_contact.py

```diff
@@ -42,6 +42,34 @@ def get_partner(self, email):
                 ("opt_out", "!=", True)
             ], limit=1).id
 
+    @api.multi
+    def write(self, values):
+        out = super().write(values)
+        # can't be simplified because we are looking for is_email_valid == False
+        # AND is_email_valid is in values (return None otherwise)
+        if values.get('is_email_valid') is False:
+            for mailing_contact in self:
+                # assure is_valid_email as not been reset (compute field might have been trigger)
+                if mailing_contact.is_email_valid:
+                    mailing_contact.is_email_valid = False
+
+        return out
+
+    @job(default_channel="root.mass_mailing_switzerland.update_partner_mailchimp")
+    @api.model
+    def update_partner_merge_fields_job(self, partner_id):
+        """Update one contact merge fields.
+        Preferred approach to avoid overloading the network.
+        :return: True or False
+        """
+        mailing_contact = self.search([("partner_id", "=", partner_id)], limit=1)
+        if mailing_contact:
+            try:
+                mailing_contact.action_update_to_mailchimp()
+            except:
+                return False
+        return True
```

Comment:
Method `update_all_merge_fields_job` just below should use your new methods and launch a job for each contact instead of doing the whole loop. The whole purpose of the task was to split this job otherwise we will have difficulties to relaunch it when we have an error occurring at the middle of the sync.